realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -538,6 +546,19 @@ public double average(String fieldName) {
         }
     }
 
+    /**
+     * Returns a distinct set of objects of a specific class. As a Realm is unordered, it is
+     * undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return a non-null {@link RealmResults} containing the distinct objects.
+     * @throws IllegalArgumentException if a field name does not exist.
+     * @throws IllegalArgumentException if a field's type is not supported.
+     * @throws UnsupportedOperationException if a field is not indexed.
+     */
+    public RealmResults<E> distinct(String fieldName) {
+        return this.where().distinct(fieldName);
```

Comment:
Edit: Actually this implementation might be okay since it is returning the distinct view. If it modified in place it would make more sense to return `void`. Just leave this as it is.